### ISSUE TITLE
createIfNotExist() methos

### ISSUE THREAD

**Issue by ramanbranavitski** [realm/realm-java#900]
It will be good if you add such method to library

**Comment by cmelchior** [comment]
Hi @ramanbranavitski 
Have you looked at our Realm.copyToRealmOrUpdate() method: http://realm.io/docs/java/0.79.0/api/io/realm/Realm.html#copyToRealmOrUpdate-E-, it does pretty much what you request.

Or do you have some other use case in mind?

**Comment by ramanbranavitski** [comment]
hmm, nice, can i ignore some fields to be update? For example i have class

public class Person(){
   private String name;
   private boolean isRegistred;
}

using this method all my field will update. How can i protect field
"isRegistred" to be updated, but all other fields should be updated.


Thanks for answer.

2015-02-24 11:35 GMT+03:00 Christian Melchior <notifications@github.com>:

**Comment by cmelchior** [comment]
No, that isn't possible, all the properties are updated. If you want to avoid field updates, that implies that your object is already in the database in which case it is easier to query for it and update relevant fields.

**Comment by ramanbranavitski** [comment]
Thanks, so it will be great to add something like
http://ormlite.com/javadoc/ormlite-core/com/j256/ormlite/dao/Dao.html#createIfNotExists(T)
to your library. Very usefull method. For example i get json and wand to
save only object that havent in database, but ignore existing.

**Comment by cmelchior** [comment]
Hmm, I can see why it could be useful in some cases. Thank you for the suggestion.

**Comment by Memorion** [comment]
+1 for this feature

**Comment by savelyevdm** [comment]
+1

**Comment by Zhuinden** [comment]
``` java
    if(realm.where(Blah.class).equalTo("id", id).count() <= 0) {
        realm.insertOrUpdate(blah);
    }
```
